fvm/environment/event_emitter.go

```diff
@@ -115,37 +112,38 @@ func (emitter *eventEmitter) EmitEvent(event cadence.Event) error {
 
 	payloadSize := uint64(len(payload))
 
-	// skip limit if payer is service account
-	if emitter.payer != emitter.chain.ServiceAddress() {
-		totalSize := emitter.eventCollection.TotalByteSize() + payloadSize
-		if totalSize > emitter.eventCollectionByteSizeLimit {
-			return errors.NewEventLimitExceededError(
-				totalSize,
-				emitter.eventCollectionByteSizeLimit)
-		}
-	}
-
```

Comment:
A little behavior change here: original behavior actually did a pre-metering. It checked sizes _before_ emit an event. The new behavior utilizes the error returned from new interface, and returns error immediately. Please confirm if this will expose some risks

Reply:
This change is fine, since we throw away all events in case of event limit hit, so it doesn't really matter if we add the last one to the list or not.